.github/workflows/upstream-dev-ci.yaml

```diff
@@ -41,7 +41,7 @@ jobs:
           conda list
       - name: Run Tests
         run: |
-          python -m pytest --verbose -rf > output-${{ matrix.python-version }}-log
+          python -m pytest -rf | tee output-${{ matrix.python-version }}-log
```

Comment:
Using `tee` here breaks the next step in which we check the status code: `failure()&&....` By using `tee`, the status code for this step will always be `0` which means that artifacts upload step will always be skipped.... For the next steps in the `upstream` job and the dependent `report` job to function properly, we need to come up with a way to signal to them that there were test failures....

Reply:
Also, this means that for push/pull request events, the upstream-dev CI check will always pass